aexpect/remote_door.py

```diff
@@ -242,8 +242,8 @@ def run_subcontrol(session, control_path, timeout=600, detach=False):
         session.set_output_func(logging.info)
         session.set_output_params(())
         session.sendline(cmd)
-    else:
-        session.cmd(cmd, timeout=timeout, print_func=logging.info)
+        return None
+    return session.cmd(cmd, timeout=timeout, print_func=logging.info)
```

Comment:
Makes sense, what would you say about using mock to replace `remote.copy_files_to` from line `242` in `remote_door` for `shutil.copy` (or open&write the local file) and using an actual `aexpect.ShellSession("sh")` for the selftest? (which is how I actually tested this on my machine using the `test_remote_door.py`)

Reply:
I undertook some steps so feel free to tell me if it reflects your points in the following commit (to be pushed):
```
commit dfec762c36dba01355bcc23ce114ab4b57a2703b (HEAD -> door-improvements)
Author: Plamen Dimitrov <pdimitrov@pevogam.com>
Date:   Mon Oct 5 13:10:00 2020 +0300

    Improved the coverage of remote door's unit tests
    
    In particular we:
    
    1) Include testing of indentation and nested functions.
    2) Make the remote mock replacements more explicit.
    3) Make use of a localized RemoteSession instance.
    
    Signed-off-by: Plamen Dimitrov <pdimitrov@pevogam.com>
```

However, I am not sure what benefit we get from mocking `remote.copy_files_to` through `shutil.copy`. We are not validating against any copied files and only check the locally stored file for the right content.